vllm/spec_decode/spec_decode_worker.py

```diff
@@ -417,12 +431,40 @@ def _run_no_spec(self, execute_model_req: ExecuteModelRequest,
                 self.previous_hidden_states.update(
                     execute_model_req.seq_group_metadata_list, hidden_states)
 
+        if self.disable_logprobs:
+            # We are skipping adding logprobs to the result. Since we are
+            # skipping the logprobs, the sampler has not serialized any
+            # tensors like sampled_token_ids to the cpu. We create the
+            # CompletionSequenceGroupOutput from the GPU tensors.
+            seq_ids = get_all_seq_ids(
+                execute_model_req.seq_group_metadata_list)
+            sampled_token_ids_list = sampler_output.sampled_token_ids.tolist()
+            completion_seq_group_output_list: List[
+                CompletionSequenceGroupOutput] = []
+            for index, seq_id in enumerate(seq_ids):
+                completion_seq_group_output_list.append(
+                    create_sequence_group_output(
+                        token_id=sampled_token_ids_list[index][0],
+                        token_id_logprob_rank=-1,
+                        token_id_logprob=0.0,
+                        seq_id=seq_id,
+                        topk_token_ids=[],
+                        topk_logprobs=[],
+                    ))
+            sampler_output_to_return = SamplerOutput(
+                outputs=completion_seq_group_output_list)
+        else:
+            # We are not skipping the logprobs. We have configured
+            # the sampler to already synchronize the GPU tensors to the
+            # CPU. Hence use the SamplerOutput directly.
+            sampler_output_to_return = sampler_output
```

Comment:
Can we move this to its own function, or a util method?
```python3
sampler_output_to_return = (self._serialize_sampler_output_no_logprobs(
    ...) if self.disable_logprobs else sampler_output)
```

Reply:
Done.